src/app/models/test-result-detail.model.ts

```diff
@@ -0,0 +1,39 @@
+export class TestResultDetail {
+  artifactId?: number;
+  buildId?: number;
+  completedOn?: Date;
+  message?: string;
+  name?: string;
+  startedOn?: Date;
+  status?: TestResultStatus;
+  testResultDetailId?: number;
+
+  constructor(
+    artifactId?: number,
+    buildId?: number,
+    completedOn?: Date,
+    message?: string,
+    name?: string,
+    startedOn?: Date,
+    status?: TestResultStatus,
+    testResultDetailId?: number) {
+
+    this.artifactId = artifactId;
+    this.buildId = buildId;
+    this.completedOn = completedOn;
+    this.message = message;
+    this.name = name;
+    this.startedOn = startedOn;
+    this.status = status;
+    this.testResultDetailId = testResultDetailId;
+  }
+}
+
+/* eslint-disable @typescript-eslint/naming-convention */
+export type TestResultStatus = 'Failed' | 'Passed' | 'Skipped';
+export const TestResultStatus = {
+  Failed: 'Failed' as TestResultStatus,
+  Passed: 'Passed' as TestResultStatus,
+  Skipped: 'Skipped' as TestResultStatus,
+};
```

Comment:
I looked at the definition for `TestResultDetail` over at wharf-api now while writing this though, and it seems I accidentally put `enums` instead of `enum` for the tag there. Generating with the proper `enum` tag made `MainTestResultDetail`'s status field be type `string` of `MainTestResultStatus`, as it should (I think, matches the `MainProvider`'s behavior).

Also looked over the usage of the defined types, and there's no real need to define them again. I think I needed to do it because of my typo with `enum` for `MainTestResultDetail`, but for some reason did it for all. Probably confusion.

Anyways, I removed those types and replaced the references with references to the types from `api-client`, which are prefixed with `Main`.

I will add the `enums` -> `enum` change to my fix-PR over at wharf-api.